src/mbgl/renderer/render_source.hpp

```diff
@@ -76,6 +76,16 @@ class RenderSource : protected TileObserver {
 
     void setObserver(RenderSourceObserver*);
 
+    class FeatureExtension {
```

Comment:
Would this extension be used for more than just a single `query` method? If not, could this be renamed to `FeatureQueryExtension` or removed entirely and the multiple extensions be managed in the subclasses?

Reply:
I removed intermediate object for simplicity. I have a question about proposal:
https://github.com/mapbox/mapbox-gl-native/pull/12726#issuecomment-435542608
```c++
mbgl::Value queryFeatureExtensions(
    std::string sourceId, <-- Source  / RenderSource
    optional<std::string> sourceLayer,
    const mbgl::Feature& feature,
    std::string extension, <-- Extension
    std::string extensionField );
}
```
It implies that one source might have multiple extensions, right? That made me thinking that it might be easier to manage separate extension for particular source in a separate implementation. So that render can do something like:

```c++
if (FeatureExtension* ext = renderSource->getExtension(extensionID)) {
   return extension->queryFeatureExtension(...);
}
```

Yet, that is an implementation detail, it can be changed if needed. Public facing interface is not affected in any way.